packages/react-core/src/components/DualListSelector/DualListSelector.tsx

```diff
@@ -535,10 +535,10 @@ export class DualListSelector extends React.Component<DualListSelectorProps, Dua
       : this.isChecked(treeItem, isChosen);
 
   mapChecked = (item: DualListSelectorTreeItemData, isChosen: boolean): DualListSelectorTreeItemData => {
-    const hasCheck = this.areAllDescendantsChecked(item, isChosen);
+    const hasPartialCheck = this.areAllDescendantsChecked(item, isChosen);
     item.isChecked = false;
 
-    if (hasCheck) {
+    if (hasPartialCheck) {
```

Comment:
Ah apologies, I think I misspoke in my previous comment. These hasPartialCheck variables on lines 538 and 541 can just stay `hasCheck`. I meant to only mention `hasPartialCheck` to compare the similarity.

Line 544 where `const hasPartialCheck` is inside the `else` block can be left as is.

Reply:
Ah my bad, I see now that the partial is based on `areSomeDescendantsChecked` as opposed to `areAll....`, should be updated now